datafusion/src/physical_plan/projection.rs

```diff
@@ -185,6 +185,10 @@ impl ExecutionPlan for ProjectionExec {
             self.expr.iter().map(|(e, _)| Arc::clone(e)),
         )
     }
+
+    fn should_repartition_children(&self) -> bool {
+        false
```

Comment:
A processing heavy projection (e.g. parsing during cast, complex case/when statement, etc.) benefits from having multiple partitions, to run in parallel.